File: module-contracts/src/main/java/com/beat/contracts/auth/JwtTokenPort.java
package com.beat.contracts.auth;

public interface JwtTokenPort {

	String issueAccessToken(JwtSubject subject);

	String issueRefreshToken(JwtSubject subject);

	TokenValidationResult validateAccessToken(String token);

	TokenValidationResult validateRefreshToken(String token);

	Long getMemberId(String token, JwtTokenType expectedType);

	String getRoleName(String token, JwtTokenType expectedType);
}

File: module-contracts/src/main/kotlin/com/beat/contracts/auth/JwtTokenPort.kt
package com.beat.contracts.auth


interface JwtTokenPort {

    fun issueAccessToken(subject: JwtSubject): String

    fun issueRefreshToken(subject: JwtSubject): String

    fun validateAccessToken(token: String): TokenValidationResult

    fun validateRefreshToken(token: String): TokenValidationResult

    fun getMemberId(token: String, expectedType: JwtTokenType): Long?

    fun getRoleName(token: String, expectedType: JwtTokenType): String?
}
